components/org.wso2.micro.integrator.initializer/src/main/java/org/wso2/micro/integrator/initializer/ServiceBusInitializer.java

```diff
@@ -165,6 +173,24 @@ protected void activate(ComponentContext ctxt) {
 
                 synapseEnvironment.registerSynapseHandler(new SynapseExternalPropertyConfigurator());
                 synapseEnvironment.registerSynapseHandler(new ProxyLogHandler());
+
+                // Register internal transaction synapse handler
+                boolean transactionPropertyEnabled = TransactionCountHandlerComponent.isTransactionPropertyEnabled();
+                if (transactionPropertyEnabled) {
+                    TransactionCountHandler transactionCountHandler = new TransactionCountHandler();
+                    synapseEnvironment.registerSynapseHandler(transactionCountHandler);
```

Comment:
Seems transactionCountHandlerComponent.start doesn't need the handler , please correct me if I'm wrong. In that case shall we just register after starting it. Then we don't need to remove on error.

Reply:
+1. fixed.